hack/pin-dependency.sh

```diff
@@ -58,41 +59,43 @@ trap "cleanup" EXIT SIGINT
 cleanup
 mkdir -p "${_tmp}"
 
-# Add the require directive
-echo "Running: go get ${dep}@${sha}"
-go get -d "${dep}@${sha}"
-
-# Find the resolved version
-rev=$(go mod edit -json | jq -r ".Require[] | select(.Path == \"${dep}\") | .Version")
-
-# No entry in go.mod, we must be using the natural version indirectly
-if [[ -z "${rev}" ]]; then
-  # backup the go.mod file, since go list modifies it
-  cp go.mod "${_tmp}/go.mod.bak"
-  # find the revision
-  rev=$(go list -m -json "${dep}" | jq -r .Version)
-  # restore the go.mod file
-  mv "${_tmp}/go.mod.bak" go.mod
+# Specifying a different repo is optional.
+replacement=
+case $dep in
+    *=*)
+        replacement=$(echo "$dep" | sed -e 's/.*=//')
+        dep=$(echo "$dep" | sed -e 's/=.*//')
```

Comment:
nit: use `${dep}`, and check that replacement and dep are not empty here?

Reply:
I've moved the parsing before the help message and extended the if check. It now does:
```
$ hack/pin-dependency.sh foobar.com/xyz= master
Usage:
  hack/pin-dependency.sh $MODULE[=$REPLACEMENT] $SHA-OR-TAG

Examples:
  hack/pin-dependency.sh github.com/docker/docker 501cb131a7b7
  hack/pin-dependency.sh github.com/docker/docker=github.com/johndoe/docker my-experimental-branch

Replacing with a different repository is useful for testing but
the result should never be merged into Kubernetes!
```